packages/svelte/src/index-server.js

```diff
@@ -1,33 +1,37 @@
-import { current_component_context } from './internal/client/runtime.js';
-
-export {
-	createEventDispatcher,
-	flushSync,
-	getAllContexts,
-	getContext,
-	hasContext,
-	mount,
-	hydrate,
-	setContext,
-	tick,
-	unmount,
-	untrack
-} from './index-client.js';
-
-/** @returns {void} */
-export function onMount() {}
+import { current_component } from './internal/server/context.js';
+import { noop } from './internal/shared/utils.js';
 
 /** @param {() => void} fn */
 export function onDestroy(fn) {
-	const context = /** @type {import('#client').ComponentContext} */ (current_component_context);
-	(context.ondestroy ??= []).push(fn);
+	var context = /** @type {import('#server').Component} */ (current_component);
+	(context.d ??= []).push(fn);
+}
+
+export {
+	noop as beforeUpdate,
+	noop as afterUpdate,
+	noop as onMount,
+	noop as flushSync,
+	run as untrack
+} from './internal/shared/utils.js';
+
+export function createEventDispatcher() {
+	return noop;
 }
 
-/** @returns {void} */
-export function beforeUpdate() {}
+export function mount() {
+	throw new Error('mount(...) is not available on the server');
+}
```

Comment:
Previously it was possible to use client APIs and its runtime in a server context, assuming the Dom APIs are shimmed. This most likely happens when using test runners. Are we ok with this use case no longer being possible? (The test runner case would need adjustments for export conditions, if that doesn't happen already automatically)

Reply:
I think this approach will lead to less confusion in general — if people expect `mount` to work without any configuration changes to their test runner or whatever, then they will probably expect things like `unstate` to work (which it won't), and things are very likely to break in future. This way, we force people to do things the right way from the get-go